### ISSUE TITLE
Incorrect device placement on colab tpu with example notebook

### ISSUE THREAD

**Issue by hadaev8** [huggingface/accelerate#181]
Here is copy of example notebook i launched
https://colab.research.google.com/drive/1xgtoMdGwHHzV3c5vWmrywKyWnl4UgZip?usp=sharing

With
```
    print(accelerator.is_main_process)
    print(accelerator.device)
```

I'm getting

```
True
xla:1
False
xla:0
False
xla:0
False
xla:0
False
xla:0
False
xla:0
False
xla:0
False
xla:0
```

On my own script, all devices xla:0 and non is the main process.
Here for some reason 2 devices used.

**Comment by sgugger** [comment]
That is not a bug, the problem is at the PyTorch XLA level where the ordinal in the device is random like this. The only integer you can trust is the `accelerator.process_index`.

**Comment by hadaev8** [comment]
@sgugger 
So, does an example notebook work correctly?
It works like x2 slower than the same script on GPU.

**Comment by sgugger** [comment]
Yes that example works correctly. Note that this is a short training, and that on TPU, the first iteration is very slow since XLA compiles the graph of computation. To see the benefits of TPU, you need to launch a longer training.

**Comment by hadaev8** [comment]
Can I trust accelerator.device ?

**Comment by sgugger** [comment]
I am not sure what you mean. Like I said, if the printed results don't satisfy you, please reach out to the PyTorch XLA team. I already flagged this issue to them but it's not a bug in their minds.

**Comment by hadaev8** [comment]
I need to create a label tensor.

```
device = accelerator.device
y = torch.cat([torch.ones(config['batch_size'], device=device), torch.zeros(config['batch_size'], device=device)])
```
I'm doing it like this, is it the right way?

**Comment by sgugger** [comment]
Yes, that's definitely the right way!

**Comment by hadaev8** [comment]
Thanks

**Comment by hadaev8** [comment]
@sgugger 
I tried again nlp example tpu vs gpu for 100 epochs, tpu seems to be like ~30% faster (hour vs hour and half), seems too slow to me.
What speed difference I should expect?
